x/dex/keeper/price.go

```diff
@@ -0,0 +1,61 @@
+package keeper
+
+import (
+	"encoding/binary"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
+)
+
+func (k Keeper) SetPriceState(ctx sdk.Context, price types.Price, contractAddr string, epoch uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PricePrefix(contractAddr))
+	b := k.Cdc.MustMarshal(&price)
+	store.Set(GetKeyForPriceState(epoch, *price.Pair), b)
+}
+
+func (k Keeper) DeletePriceState(ctx sdk.Context, contractAddr string, epoch uint64, pair types.Pair) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PricePrefix(contractAddr))
+	store.Delete(GetKeyForPriceState(epoch, pair))
+}
+
+func (k Keeper) GetPriceState(ctx sdk.Context, contractAddr string, epoch uint64, pair types.Pair) (types.Price, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PricePrefix(contractAddr))
+	res := types.Price{}
+	key := GetKeyForPriceState(epoch, pair)
+	if !store.Has(key) {
+		res.Pair = &pair
+		return res, false
+	}
+	b := store.Get(key)
+	k.Cdc.MustUnmarshal(b, &res)
+	return res, true
+}
+
+func (k Keeper) GetAllPrices(ctx sdk.Context, contractAddr string, pair types.Pair) (list []*types.Price) {
```

Comment:
when do we purge prices we no longer need?

Reply:
https://github.com/sei-protocol/sei-chain/pull/19/files#diff-f32c70484252fadec70cda9a6ad7888fd568f6afffb3750bb9621090e2ac88bbR249